src/main/java/dev/binclub/javaception/classfile/ClassFileParser.java

```diff
@@ -137,6 +142,45 @@ private ClassFileParser(byte[] data, int startOffset, InstanceOop loader) throws
 		}
 		
 		readClassAttributes(attributesOffset, constantPool);
+		sortFields();
+	}
+	
+	public static Klass parse(InputStream stream, InstanceOop loader) throws IOException, ClassNotFoundException {
+		return new ClassFileParser(stream.readAllBytes(), loader).toKlass();
+	}
+	
+	public static Klass parse(ByteBuffer buffer, InstanceOop loader) throws ClassNotFoundException {
+		byte[] data = new byte[buffer.remaining()];
+		buffer.get(data);
+		return new ClassFileParser(data, loader).toKlass();
+	}
+	
+	public static Klass parse(byte[] data, InstanceOop loader) throws ClassNotFoundException {
+		return new ClassFileParser(data, loader).toKlass();
+	}
+	
+	//fields are sorted so non static are first makes indexing easier
+	public void sortFields() {
+		List<FieldInfo> tempFields = new ArrayList<>();
+		Collections.addAll(tempFields, fields);
+		tempFields.sort((f1, f2) -> {
+			boolean f1Static = (f1.access & ACC_STATIC) != 0;
+			boolean f2Static = (f2.access & ACC_STATIC) != 0;
+			return Boolean.compare(f1Static, f2Static);
+		});
+		tempFields.toArray(fields);
```

Comment:
I would recommend making this `fields = tempFields.toArray(fields);` because `toArray` does not always modify the provided array, it can provide a copy.